### ISSUE TITLE
[Error] Fail to execute `devenv build qt`

### ISSUE THREAD

**Issue by ThreeMonth03** [solvcon/devenv#166]
Here is the error log:
```bash
zhangjunshideMacBook-Air:flavors abao$ devenv build qt
Execute building script "/Users/abao/vscode/devenv/scripts/build.d/qt"  ...
qt-6.6.3.tar.xz md5 hash 0e2c9dd87cbc6768da2bfc7f776c272f confirmed
//Users/abao/vscode/devenv/flavors
~/vscode/devenv/flavors/bad/src/qt-6.6.3 //Users/abao/vscode/devenv/flavors
error: can't open patch '/Users/abao/vscode/devenv/var/patch/qt-6.6.3.patch': No such file or directory
//Users/abao/vscode/devenv/flavors
~/vscode/devenv/flavors/bad/src/qt-6.6.3/build //Users/abao/vscode/devenv/flavors
run command: cmake -DCMAKE_INSTALL_PREFIX=/Users/abao/vscode/devenv/flavors/bad/usr -DCMAKE_BUILD_TYPE=Release -DBUILD_qtquicktimeline=OFF -DBUILD_qtquick3d=OFF -DBUILD_qt5compat=OFF -DBUILD_qtactiveqt=OFF -DBUILD_qtcharts=OFF -DBUILD_qtcoap=OFF -DBUILD_qtconnectivity=OFF -DBUILD_qtdatavis3d=OFF -DBUILD_qtwebsockets=OFF -DBUILD_qthttpserver=OFF -DBUILD_qttools=OFF -DBUILD_qtdoc=OFF -DBUILD_qtlottie=OFF -DBUILD_qtmqtt=OFF -DBUILD_qtnetworkauth=OFF -DBUILD_qtopcua=OFF -DBUILD_qtserialport=OFF -DBUILD_qtlocation=OFF -DBUILD_qtpositioning=OFF -DBUILD_qtquick3dphysics=OFF -DBUILD_qtremoteobjects=OFF -DBUILD_qtscxml=OFF -DBUILD_qtsensors=OFF -DBUILD_qtserialbus=OFF -DBUILD_qtspeech=OFF -DBUILD_qttranslations=OFF -DBUILD_qtvirtualkeyboard=OFF -DBUILD_qtwayland=OFF -DBUILD_qtwebchannel=OFF -DBUILD_qtwebengine=OFF -DBUILD_qtwebview=OFF -DBUILD_qtquickeffectmaker=OFF -DBUILD_qtgrpc=OFF -DBUILD_qtmultimedia=OFF -DQT_ALLOW_SYMLINK_IN_PATHS=ON -DCMAKE_PREFIX_PATH=/Users/abao/vscode/devenv/flavors/bad/usr -G Ninja ..
-- The CXX compiler identification is AppleClang 15.0.0.15000309
-- The C compiler identification is AppleClang 15.0.0.15000309
-- The ASM compiler identification is AppleClang
-- Found assembler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Checking dependencies of submodule 'qt-everywhere-src-6.6.3'
-- Checking dependencies of submodule 'qtbase'
-- Checking dependencies of submodule 'qtimageformats'
-- Checking dependencies of submodule 'qtlanguageserver'
-- Checking dependencies of submodule 'qtshadertools'
-- Checking dependencies of submodule 'qtsvg'
-- Checking dependencies of submodule 'qtdeclarative'
-- Checking dependencies of submodule 'qt3d'
-- Skipping optional dependency 'qtmultimedia' of 'qt3d', because building 'qtmultimedia' was explicitly disabled.
-- Checking dependencies of submodule 'qtgraphs'
CMake Error at CMakeLists.txt:79 (message):
  Module 'qtgraphs' depends on 'qtquick3d', but building 'qtquick3d' was
  explicitly disabled.

  Note: Use '-skip qtgraphs' to exclude it from the build.


-- Configuring incomplete, errors occurred!
done; log file: /Users/abao/vscode/devenv/flavors/bad/src/qt-6.6.3/build/configure.log
run command: cmake --build . --parallel
ninja: error: loading 'build.ninja': No such file or directory
done; log file: /Users/abao/vscode/devenv/flavors/bad/src/qt-6.6.3/build/cmake.log
run command: cmake --install .
CMake Error: Not a file: /Users/abao/vscode/devenv/flavors/bad/src/qt-6.6.3/build/cmake_install.cmake
CMake Error: Error processing file: /Users/abao/vscode/devenv/flavors/bad/src/qt-6.6.3/build/cmake_install.cmake
done; log file: /Users/abao/vscode/devenv/flavors/bad/src/qt-6.6.3/build/install.log
//Users/abao/vscode/devenv/flavors
Finished building script "/Users/abao/vscode/devenv/scripts/build.d/qt".
```

If we comment the following line and enable the installation of 'qtquick3d', the script will work regularly.
https://github.com/solvcon/devenv/blob/ada270f51856644ea937581072eb8a43e5c54c84/scripts/build.d/qt#L76

**Comment by ThreeMonth03** [comment]
It is suitable for newcomer to make a pull request, and my friend will solve the issue.

**Comment by yungyuc** [comment]
Out of curiosity.  How much time it took you to successfully build the qt from source?

**Comment by ThreeMonth03** [comment]
> Out of curiosity. How much time it took you to successfully build the qt from source?

I remembered that it took me about 30 minutes to build qt on MacBook Air M1.

**Comment by yungyuc** [comment]
> > Out of curiosity. How much time it took you to successfully build the qt from source?
> 
> I remembered that it took me about 30 minutes to build qt on MacBook Air M1.

I think it's optimized/release build right?  That's similar to what I felt too.  Apple Silicons are fast.

**Comment by ThreeMonth03** [comment]
> > > Out of curiosity. How much time it took you to successfully build the qt from source?
> > 
> > 
> > I remembered that it took me about 30 minutes to build qt on MacBook Air M1.
> 
> I think it's optimized/release build right? That's similar to what I felt too. Apple Silicons are fast.

Yes, it is. the script downloads the [official software release](https://download.qt.io/official_releases/qt/) and builds it locally.
https://github.com/solvcon/devenv/blob/ada270f51856644ea937581072eb8a43e5c54c84/scripts/build.d/qt#L28
https://github.com/solvcon/devenv/blob/ada270f51856644ea937581072eb8a43e5c54c84/scripts/build.d/qt#L129-L131

**Comment by yungyuc** [comment]
When we say "optimized/release" build, it usually means the build is not a debug build.  We are not referring to the source release.

**Comment by ThreeMonth03** [comment]
> When we say "optimized/release" build, it usually means the build is not a debug build. We are not referring to the source release.

OK, I misunderstand the "release" build. It is not a debug build.
https://github.com/solvcon/devenv/blob/ada270f51856644ea937581072eb8a43e5c54c84/scripts/build.d/qt#L70-L71

**Comment by ThreeMonth03** [comment]
Fixed by pr #167.

**Comment by yungyuc** [comment]
Good work, @ThreeMonth03 and @dragonwu0919
